community/io/src/test/java/org/neo4j/adversaries/pagecache/AdversarialWritePageCursor.java

```diff
@@ -264,7 +264,11 @@ public boolean shouldRetry() throws IOException
     public int copyTo( int sourceOffset, PageCursor targetCursor, int targetOffset, int lengthInBytes )
     {
         adversary.injectFailure( IndexOutOfBoundsException.class );
-        return delegate.copyTo( sourceOffset, targetCursor, targetOffset, lengthInBytes );
+
+        PageCursor targetCursorDelegate = targetCursor instanceof AdversarialWritePageCursor ?
```

Comment:
I believe this was the recommended fix after discussing briefly with @chrisvest although I do see that there's another PR now which introduces a method that can be used instead of doing the instanceof check inside MuninnWritePageCursor itself, right. I'm thinking of https://github.com/neo4j/neo4j/pull/8540